jetbrains-core/src/software/aws/toolkits/jetbrains/core/tools/ToolType.kt

```diff
@@ -32,6 +38,17 @@ interface ToolType<VersionScheme : Version> {
      * Returns the [Version] for the executable of this type located at the specified location
      */
     fun determineVersion(path: Path): VersionScheme
+
+    companion object {
+        internal val EP_NAME = ExtensionPointName.create<DocumentedToolType<Version>>("aws.toolkit.tool")
```

Comment:
Seems a little odd to me that the EP is tied to the DocumentedToolType instead of ToolType, and a  `is` check is done in the configurable.

But I can see the appeal and easy to change if the need ever arises

Reply:
Yeah I think there may be cases where we don't want to allow customers to override and then it doesn't need to be registered in the EP (and thus won't be configurable), however I do think we want docs in the event it is surfaced